### DISCUSSION TITLE
'extracting' Psychrometrics.cc/hh into a stand alone *.dll

### DISCUSSION THREAD

**Discussion by TfedUD** [NREL/EnergyPlus#9178] [General]
Hello! cpp isn't really (at all..) my language (py guy). so I'm not really sure if this could work but I had an idea and felt nothing to loose but ask:

I'd like to if possible (respecting and following the correct licensing processes and procedures):
 'extract' the Psychrometrics.cc/hh  via some git actions/workflows and compile into a stand alone *.dll to facilitate an E+ based python package, that would stay up to date with E+ dev:
 
 Not sure on what all 'imports' are explicitly for functionality with the entirety of E+ functionality and could theoretically be removed for this purpose:
    nore if this would even be possible in terms of 'basically copy/pasting with some easy edits' to create a stand alone *.dll from  E+ Psychometrics as is in mind.
    
 Thanks for any one's time and knowledge!
 @TfedUD

**Comment by mitchute** [comment]
Can't you just use CoolProp? https://pypi.org/search/?q=CoolProp

**Reply by TfedUD** [reply]
because I didn't know it existed until now! Ty!
Also kinda just wanted to make a package myself that piggybacks on E+:
   CoolProp should keep me busy enough with fun for the time being tho! 
Thank you!

**Reply by Myoldmopar** [reply]
EnergyPlus also already comes as a callable DLL when you install it.  You can access all the psychrometric functions by linking to the DLL in the install folder.  See our tests for examples: https://github.com/NREL/EnergyPlus/blob/develop/tst/EnergyPlus/api/TestFunctional.c#L105

**Reply by TfedUD** [reply]
```facepalm``` Thank you! I totally forgot that could all be accessed 'stand alone'
